src/System.Private.CoreLib/shared/System/Threading/ThreadPool.cs

```diff
@@ -633,6 +641,8 @@ internal static bool Dispatch()
                         Unsafe.As<IThreadPoolWorkItem>(workItem).Execute();
                     }
 
+                    Thread.CurrentThread.ResetThreadPoolThread();
```

Comment:
What's this do?

Reply:
In CoreRT it resets the name of the thread, apartment state and priority, or generally anything that the user can change about the  thread pool thread. In CoreCLR this is currently part of the C++ thread pool code.

Reply:
/Fixed the unnecessary access to thread static variable, thanks./

`Thread::InternalReset` gets called as part of `NotifyWorkItemComplete`, which is just few lines below in the same loop. It's matter of correctness to run it for each iteration where a user code is involved. Otherwise the next user code could accidentally run in a high-priority thread or some other unpredictable scenario. In most cases it just checks whether the state is intact and doesn't do any expensive state change.

I plan to move it out of the unmanaged code to `ResetThreadPoolThread` if I can confirm that there's no performance impact.